server/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSourceConfig.java

```diff
@@ -73,51 +85,55 @@ public static  ValuesSourceConfig resolve(
             }
             /*
              * This is the Stand Alone Script path.  We should have a script that will produce a value independent of the presence or
-             * absence of any one field.  The type of the script is given by the valueType field, and defaults to bytes if not specified.
+             * absence of any one field.  The type of the script is given by the userValueTypeHint field, and defaults to bytes if not
+             * specified.
              */
-            // TODO: Not pluggable, should always be valueType if specified, BYTES if not.
+            // TODO: Not pluggable, should always be userValueTypeHint if specified, BYTES if not.
             // TODO: Probably should validate that the resulting type is valid for this agg.  That needs to be plugable.
-            ValuesSourceType valuesSourceType = valueType != null ? valueType.getValuesSourceType() : CoreValuesSourceType.ANY;
+            valuesSourceType = userValueTypeHint != null ? userValueTypeHint.getValuesSourceType() : CoreValuesSourceType.ANY;
```

Comment:
There's no reason to bounce through an ANY here; this should either be what the user specified or the default.